app/assets/javascripts/discourse/components/plugin-connector.js.es6

```diff
@@ -45,6 +57,19 @@ export default Component.extend({
     connectorClass.setupComponent.call(this, merged, this);
   },
 
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    this._decoratePluginOutlets();
+  },
+
+  @afterRender
+  _decoratePluginOutlets() {
+    _decorators
+      .filterBy("connector", this.connector.outletName)
```

Comment:
You could replace this `filterBy` with a hash lookup based on the outlet name. 

for example if decorators looked like:

`{ "my-outlet-name": [ dec0, dec1, dec2 ] }`

You could then:

`(_decorators[this.connector.outletName] || []) .forEach(dec => dec.callback(this.element, this.args))`